doc/api/icu.md

```diff
@@ -0,0 +1,76 @@
+# icu
+
+    Stability: 1 - Experimental
+
+The `icu` module provides an interface to Unicode and Internationalization
+functionality provided by the ICU4C library.
+
+```js
+const icu = require('icu');
```

Comment:
ICU includes charset detection APIs that I had in here originally but opted to pull back out based on the feedback.